proxy/api/src/project.rs

```diff
@@ -71,6 +75,76 @@ where
     }
 }
 
+/// A Radicle project that you're interested in but haven't contributed to.
+///
+/// See [`Projects`] for a detailed breakdown of both kinds of projects.
+#[derive(Serialize)]
+pub struct Tracked(Project);
+
+impl Deref for Tracked {
+    type Target = Project;
+
+    fn deref(&self) -> &Self::Target {
+        &self.0
+    }
+}
+
+/// A user can view projects that are in different states in their application.
+pub struct Projects {
+    /// A project that is tracked is one that the user has replicated onto their device but has not
+    /// made any changes to. A project is still considered tracked if they checked out a working copy but
+    /// have not performed any commits to the references.
+    pub tracked: Vec<Tracked>,
+    /// A project that is "mine" is one that the user has either:
+    ///     a. Created themselves using the application.
+    ///     b. Replicated (see tracked above), checked out a working copy, and push changes to
+    ///     references.
+    ///
+    /// The conditions imply that a project is "mine" if I am the maintainer or I have contributed
+    /// to the project.
+    pub mine: Vec<Project>,
+}
+
+impl Projects {
```

Comment:
Introducing this type might impede on the ergonomics, as we in all places expect a `Vec` which can easily be manipulated like calling `first`. Additionally it might not communicate as well the actual shape of the data. Would it be feasible to approach this differently with lists of:

``` rust
enum ProjectRelation {
    Contributed(Project),
    Tracked(Project),
}

fn list(api: &coco::Api) -> Vec<ProjectRelation> {
    todo!()
}
```

With the help of that the caller can normally operate on the list by filtering, mapping and whatnot.

Reply:
The caller could do `into_iter().first()`. But if they wanted to get the first of a tracked or contributed project it seems easier to do `projects.tracked.first()` over something that needs to do a `filter` followed by `first`. 

Since the idea is these things are generally separated, I'd like to try to keep it in this form. Unless you see any other strong reasons why it would be better to have the above?

Reply:
I follow your point, @FintanH, and agree with the value it provides :+1: